java/main/src/test/java/com/saucelabs/saucebindings/options/VisualOptionsTest.java

```diff
@@ -0,0 +1,269 @@
+package com.saucelabs.saucebindings.options;
+
+import com.saucelabs.saucebindings.Browser;
+import com.saucelabs.saucebindings.CITools;
+import com.saucelabs.saucebindings.GitManager;
+import com.saucelabs.saucebindings.JobVisibility;
+import com.saucelabs.saucebindings.PageLoadStrategy;
+import com.saucelabs.saucebindings.Prerun;
+import com.saucelabs.saucebindings.SaucePlatform;
+import com.saucelabs.saucebindings.SystemManager;
+import com.saucelabs.saucebindings.UnhandledPromptBehavior;
+import lombok.SneakyThrows;
+import org.junit.Test;
+import org.openqa.selenium.MutableCapabilities;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class VisualOptionsTest {
+    @Test
+    public void createsDefaultVisualOptions() {
+        VisualOptions visualOptions = new VisualOptions("Required Test Name");
+
+        MutableCapabilities visualCapabilities = new MutableCapabilities();
+        visualCapabilities.setCapability("apiKey", System.getenv("SCREENER_API_KEY"));
+        visualCapabilities.setCapability("branch", GitManager.getCurrentBranch());
+        visualCapabilities.setCapability("projectName", CITools.getBuildName());
+
+        assertEquals(visualCapabilities, visualOptions.toCapabilities().getCapability("sauce:visual"));
+    }
+
+    @Test
+    public void acceptsVisualOptions() {
+        SauceOptions sauceOptions = SauceOptions.firefox()
+                .setPlatformName(SaucePlatform.MAC_HIGH_SIERRA)
+                .setBrowserVersion("99")
+                .setName("Visual Options Needs a Name")
+                .build();
+
+        Map<String, Object> diffOptions = new HashMap<>();
+        diffOptions.put("structure", true);
```

Comment:
I find this map initialisation syntax a bit too talkative. We might use some available shortcuts instead: https://www.baeldung.com/java-initialize-hashmap

Reply:
BTW, guava is a part of selenium, so it is safe to use ImmutableMap.of()

Reply:
lol, yeah, I just started using it. This code is a copy/paste from other stuff I've done.  I'll take a look at those examples.